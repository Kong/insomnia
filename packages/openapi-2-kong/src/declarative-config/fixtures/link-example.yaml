openapi: 3.0.0
info:
  title: Link Example
  version: 1.0.0
servers:
  - url: http://backend.com/path
paths:
  /2.0/users/{username}:
    get:
      operationId: getUserByName
      parameters:
        - name: username
          in: path
          required: true
          schema:
            type: string
      responses:
        '200':
          description: The User
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/user'
          links:
            userRepositories:
              $ref: '#/components/links/UserRepositories'
  /2.0/repositories/{username}:
    get:
      operationId: getRepositoriesByOwner
      parameters:
        - name: username
          in: path
          required: true
          schema:
            type: string
      responses:
        '200':
          description: repositories owned by the supplied user
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: '#/components/schemas/repository'
          links:
            userRepository:
              $ref: '#/components/links/UserRepository'
  /2.0/repositories/{username}/{slug}:
    get:
      operationId: getRepository
      parameters:
        - name: username
          in: path
          required: true
          schema:
            type: string
        - name: slug
          in: path
          required: true
          schema:
            type: string
      responses:
        '200':
          description: The repository
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/repository'
          links:
            repositoryPullRequests:
              $ref: '#/components/links/RepositoryPullRequests'
  /2.0/repositories/{username}/{slug}/pullrequests:
    get:
      operationId: getPullRequestsByRepository
      parameters:
        - name: username
          in: path
          required: true
          schema:
            type: string
        - name: slug
          in: path
          required: true
          schema:
            type: string
        - name: state
          in: query
          schema:
            type: string
            enum:
              - open
              - merged
              - declined
      responses:
        '200':
          description: an array of pull request objects
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: '#/components/schemas/pullrequest'
  /2.0/repositories/{username}/{slug}/pullrequests/{pid}:
    get:
      operationId: getPullRequestsById
      parameters:
        - name: username
          in: path
          required: true
          schema:
            type: string
        - name: slug
          in: path
          required: true
          schema:
            type: string
        - name: pid
          in: path
          required: true
          schema:
            type: string
      responses:
        '200':
          description: a pull request object
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/pullrequest'
          links:
            pullRequestMerge:
              $ref: '#/components/links/PullRequestMerge'
  /2.0/repositories/{username}/{slug}/pullrequests/{pid}/merge:
    post:
      operationId: mergePullRequest
      parameters:
        - name: username
          in: path
          required: true
          schema:
            type: string
        - name: slug
          in: path
          required: true
          schema:
            type: string
        - name: pid
          in: path
          required: true
          schema:
            type: string
      responses:
        '204':
          description: the PR was successfully merged
components:
  links:
    UserRepositories:
      # returns array of '#/components/schemas/repository'
      operationId: getRepositoriesByOwner
      parameters:
        username: $response.body#/username
    UserRepository:
      # returns '#/components/schemas/repository'
      operationId: getRepository
      parameters:
        username: $response.body#/owner/username
        slug: $response.body#/slug
    RepositoryPullRequests:
      # returns '#/components/schemas/pullrequest'
      operationId: getPullRequestsByRepository
      parameters:
        username: $response.body#/owner/username
        slug: $response.body#/slug
    PullRequestMerge:
      # executes /2.0/repositories/{username}/{slug}/pullrequests/{pid}/merge
      operationId: mergePullRequest
      parameters:
        username: $response.body#/author/username
        slug: $response.body#/repository/slug
        pid: $response.body#/id
  schemas:
    user:
      type: object
      properties:
        username:
          type: string
        uuid:
          type: string
    repository:
      type: object
      properties:
        slug:
          type: string
        owner:
          $ref: '#/components/schemas/user'
    pullrequest:
      type: object
      properties:
        id:
          type: integer
        title:
          type: string
        repository:
          $ref: '#/components/schemas/repository'
        author:
          $ref: '#/components/schemas/user'
