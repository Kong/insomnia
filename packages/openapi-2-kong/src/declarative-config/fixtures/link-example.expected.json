{
  "services": [
    {
      "protocol": "http",
      "host": "Link_Example",
      "port": 80,
      "path": "/path",
      "name": "Link_Example",
      "plugins": [],
      "routes": [
        {
          "methods": ["GET"],
          "strip_path": false,
          "tags": ["OAS3_import", "OAS3file_link-example.yaml"],
          "name": "Link_Example-getUserByName",
          "paths": ["/2.0/users/(?<username>[^\\/\\s]+)$"]
        },
        {
          "methods": ["GET"],
          "strip_path": false,
          "tags": ["OAS3_import", "OAS3file_link-example.yaml"],
          "name": "Link_Example-getRepositoriesByOwner",
          "paths": ["/2.0/repositories/(?<username>[^\\/\\s]+)$"]
        },
        {
          "methods": ["GET"],
          "strip_path": false,
          "tags": ["OAS3_import", "OAS3file_link-example.yaml"],
          "name": "Link_Example-getRepository",
          "paths": [
            "/2.0/repositories/(?<username>[^\\/\\s]+)/(?<slug>[^\\/\\s]+)$"
          ]
        },
        {
          "methods": ["GET"],
          "strip_path": false,
          "tags": ["OAS3_import", "OAS3file_link-example.yaml"],
          "name": "Link_Example-getPullRequestsByRepository",
          "paths": [
            "/2.0/repositories/(?<username>[^\\/\\s]+)/(?<slug>[^\\/\\s]+)/pullrequests$"
          ]
        },
        {
          "methods": ["GET"],
          "strip_path": false,
          "tags": ["OAS3_import", "OAS3file_link-example.yaml"],
          "name": "Link_Example-getPullRequestsById",
          "paths": [
            "/2.0/repositories/(?<username>[^\\/\\s]+)/(?<slug>[^\\/\\s]+)/pullrequests/(?<pid>[^\\/\\s]+)$"
          ]
        },
        {
          "methods": ["POST"],
          "strip_path": false,
          "tags": ["OAS3_import", "OAS3file_link-example.yaml"],
          "name": "Link_Example-mergePullRequest",
          "paths": [
            "/2.0/repositories/(?<username>[^\\/\\s]+)/(?<slug>[^\\/\\s]+)/pullrequests/(?<pid>[^\\/\\s]+)/merge$"
          ]
        }
      ],
      "tags": ["OAS3_import", "OAS3file_link-example.yaml"]
    }
  ],
  "upstreams": [
    {
      "tags": ["OAS3_import", "OAS3file_link-example.yaml"],
      "name": "Link_Example",
      "targets": [
        {
          "target": "backend.com:80",
          "tags": ["OAS3_import", "OAS3file_link-example.yaml"]
        }
      ]
    }
  ],
  "_format_version": "1.1"
}
